airbyte-integrations/connectors/source-shopify/unit_tests/graphql_bulk/test_job.py

```diff
@@ -365,3 +365,44 @@ def test_expand_stream_slices_job_size(
     list(stream.parse_response(test_bulk_response))
     # check the next slice
     assert stream.job_manager.job_size == adjusted_slice_size
+
+@pytest.mark.parametrize(
```

Comment:
```python
@pytest.mark.parametrize(
    "job_response, error_type, expected",
    [
        (
            "bulk_job_created_response",
            ConnectionError,
            "ConnectionError(104, 'Connection was relet by peer')",
        ),
    ],
    ids=[
        "Failure after ConnectionError",
    ],
)
def test_job_healthcheck_with_connection_errors(mocker, request, requests_mock, job_response, auth_config, error_type, expected) -> None:
    stream = MetafieldOrders(auth_config)

    # Modify the sleep time for the test
    stream.job_manager.concurrent_max_retry = 1
    stream.job_manager.concurrent_interval_sec = 1
    stream.job_manager.job_check_interval_sec = 1

    # Get job_id from fixture.
    job_id = request.getfixturevalue(job_response).get("data", {}).get("node", {}).get("id")

    # Patch the method to get the right ID checks
    if job_id:
        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_get_id", value=job_id)
        # we patch the method to raise the exception, to skip the complex patching of the whole logic
        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_check_for_errors", side_effect=error_type(expected))

    # Mock the response for STATUS CHECKS
    requests_mock.post(stream.job_manager.base_url, json=request.getfixturevalue(job_response))
    # register - call 1
    test_job_status_response = requests.post(stream.job_manager.base_url)
    with pytest.raises(error_type) as error:
        # register call 2, call 3 - retried with message
        stream.job_manager.job_check(test_job_status_response)
    
    # The retried request should FAIL here, because we still want to see the Exception raised
    # We expect the call count to be 3 due to the status checks, the non-retried request would take 3 calls.
    assert expected in repr(error.value) and requests_mock.call_count == 3
```
```suggestion
@pytest.mark.parametrize(
```

Reply:
I've checked the test, and it looks fine. I've suggested my corrections to keep the methods triggered correctly, and the order is preserved, corrected the message value expected and added the check for it.